### PR TITLE
Support negative period prefix in `Calendar.ISO.parse_duration/1`

### PR DESCRIPTION
**Pull request by tfiedlerdejanze**
Negative durations are actually [specified](https://tc39.es/proposal-temporal/docs/duration.html#from) in ISO 8601-2 while individual negative units are not. The [Java Duration class](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#parse-java.lang.CharSequence-) supports both features and i think the negative duration prefix may actually be more common than the one for individual units. IMHO it makes sense that we also support both in the standard library.

### CODE REVIEW

File: lib/elixir/lib/calendar/iso.ex [hunk 1]
```diff
@@ -670,11 +670,25 @@ defmodule Calendar.ISO do
...
+      {:ok, fields} ->
+        {:ok,
+         Enum.map(fields, fn
+           {:microsecond, {value, precision}} -> {:microsecond, {-value, precision}}
+           {unit, value} -> {unit, -value}
+         end)}
+
+      error ->
+        error
+    end
```
Comment (josevalim) [inline lib/elixir/lib/calendar/iso.ex:689]:
```suggestion
    with {:ok, fields} <- parse_duration_date(string, [], year: ?Y, month: ?M, week: ?W, day: ?D) do
      {:ok,
       Enum.map(fields, fn
         {:microsecond, {value, precision}} -> {:microsecond, {-value, precision}}
         {unit, value} -> {unit, -value}
       end)}
   end
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by tfiedlerdejanze** [comment]
@josevalim given the links i posted in the PR description, would you be open to also allowing `+P10D`? I don't really see a practical need for it, but would add it since it's specified in ISO 8601-2. With that, i think we'd cover everything we want for the extended specification, except for the fact that we intentionally allow decimal fractions only for seconds

**Comment by josevalim** [comment]
Yes, we should definitely support the plus version too.

**Pull request merged**